## reanalysis the Lnc-RNA dataset

## trimming the fastq dataset

###
## all dataset
/data1/linhua/QIANLAB/PROJECT/Long-Noncoding-RNA-project/RAW-data-2016-12-07/xiaokang_lncRNA/Cleandata/BN2-972/BN2-972_R2.fq.gz
/data1/linhua/QIANLAB/PROJECT/Long-Noncoding-RNA-project/RAW-data-2016-12-07/xiaokang_lncRNA/Cleandata/BN2-972/BN2-972_R1.fq.gz
/data1/linhua/QIANLAB/PROJECT/Long-Noncoding-RNA-project/RAW-data-2016-12-07/xiaokang_lncRNA/Cleandata/BN2-971/BN2-971_R2.fq.gz
/data1/linhua/QIANLAB/PROJECT/Long-Noncoding-RNA-project/RAW-data-2016-12-07/xiaokang_lncRNA/Cleandata/BN2-971/BN2-971_R1.fq.gz
/data1/linhua/QIANLAB/PROJECT/Long-Noncoding-RNA-project/RAW-data-2016-12-07/xiaokang_lncRNA/Cleandata/BN2-973/BN2-973_R2.fq.gz
/data1/linhua/QIANLAB/PROJECT/Long-Noncoding-RNA-project/RAW-data-2016-12-07/xiaokang_lncRNA/Cleandata/BN2-973/BN2-973_R1.fq.gz
/data1/linhua/QIANLAB/PROJECT/Long-Noncoding-RNA-project/RAW-data-2016-12-07/xiaokang_lncRNA/Cleandata/BN2-1058/BN2-1058_R2.fq.gz
/data1/linhua/QIANLAB/PROJECT/Long-Noncoding-RNA-project/RAW-data-2016-12-07/xiaokang_lncRNA/Cleandata/BN2-1058/BN2-1058_R1.fq.gz
/data1/linhua/QIANLAB/PROJECT/Long-Noncoding-RNA-project/RAW-data-2016-12-07/xiaokang_lncRNA/Cleandata/BN2-1057/BN2-1057_R2.fq.gz
/data1/linhua/QIANLAB/PROJECT/Long-Noncoding-RNA-project/RAW-data-2016-12-07/xiaokang_lncRNA/Cleandata/BN2-1057/BN2-1057_R1.fq.gz
/data1/linhua/QIANLAB/PROJECT/Long-Noncoding-RNA-project/RAW-data-2016-12-07/xiaokang_lncRNA/Cleandata/BN2-1059/BN2-1059_R1.fq.gz
/data1/linhua/QIANLAB/PROJECT/Long-Noncoding-RNA-project/RAW-data-2016-12-07/xiaokang_lncRNA/Cleandata/BN2-1059/BN2-1059_R2.fq.gz
/data1/linhua/QIANLAB/PROJECT/Long-Noncoding-RNA-project/RAW-data-2016-12-07/xiaokang_lncRNA/Cleandata/BN2-1060/BN2-1060_R2.fq.gz
/data1/linhua/QIANLAB/PROJECT/Long-Noncoding-RNA-project/RAW-data-2016-12-07/xiaokang_lncRNA/Cleandata/BN2-1060/BN2-1060_R1.fq.gz
/data1/linhua/0803-LncRNA-WXK-2-JC/Cleandata/BN2-638/BN2-638_R1.fq.gz
/data1/linhua/0803-LncRNA-WXK-2-JC/Cleandata/BN2-638/BN2-638_R2.fq.gz
/data1/linhua/0803-LncRNA-WXK-2-JC/Cleandata/BN2-637/BN2-637_R1.fq.gz
/data1/linhua/0803-LncRNA-WXK-2-JC/Cleandata/BN2-637/BN2-637_R2.fq.gz

# link to a new position
ln -s /data1/linhua/QIANLAB/PROJECT/Long-Noncoding-RNA-project/RAW-data-2016-12-07/xiaokang_lncRNA/Cleandata/BN2-972/BN2-972_R2.fq.gz .
ln -s /data1/linhua/QIANLAB/PROJECT/Long-Noncoding-RNA-project/RAW-data-2016-12-07/xiaokang_lncRNA/Cleandata/BN2-972/BN2-972_R1.fq.gz .
ln -s /data1/linhua/QIANLAB/PROJECT/Long-Noncoding-RNA-project/RAW-data-2016-12-07/xiaokang_lncRNA/Cleandata/BN2-971/BN2-971_R2.fq.gz .
ln -s /data1/linhua/QIANLAB/PROJECT/Long-Noncoding-RNA-project/RAW-data-2016-12-07/xiaokang_lncRNA/Cleandata/BN2-971/BN2-971_R1.fq.gz .
ln -s /data1/linhua/QIANLAB/PROJECT/Long-Noncoding-RNA-project/RAW-data-2016-12-07/xiaokang_lncRNA/Cleandata/BN2-973/BN2-973_R2.fq.gz .
ln -s /data1/linhua/QIANLAB/PROJECT/Long-Noncoding-RNA-project/RAW-data-2016-12-07/xiaokang_lncRNA/Cleandata/BN2-973/BN2-973_R1.fq.gz .
ln -s /data1/linhua/QIANLAB/PROJECT/Long-Noncoding-RNA-project/RAW-data-2016-12-07/xiaokang_lncRNA/Cleandata/BN2-1058/BN2-1058_R2.fq.gz .
ln -s /data1/linhua/QIANLAB/PROJECT/Long-Noncoding-RNA-project/RAW-data-2016-12-07/xiaokang_lncRNA/Cleandata/BN2-1058/BN2-1058_R1.fq.gz .
ln -s /data1/linhua/QIANLAB/PROJECT/Long-Noncoding-RNA-project/RAW-data-2016-12-07/xiaokang_lncRNA/Cleandata/BN2-1057/BN2-1057_R2.fq.gz .
ln -s /data1/linhua/QIANLAB/PROJECT/Long-Noncoding-RNA-project/RAW-data-2016-12-07/xiaokang_lncRNA/Cleandata/BN2-1057/BN2-1057_R1.fq.gz .
ln -s /data1/linhua/QIANLAB/PROJECT/Long-Noncoding-RNA-project/RAW-data-2016-12-07/xiaokang_lncRNA/Cleandata/BN2-1059/BN2-1059_R1.fq.gz .
ln -s /data1/linhua/QIANLAB/PROJECT/Long-Noncoding-RNA-project/RAW-data-2016-12-07/xiaokang_lncRNA/Cleandata/BN2-1059/BN2-1059_R2.fq.gz .
ln -s /data1/linhua/QIANLAB/PROJECT/Long-Noncoding-RNA-project/RAW-data-2016-12-07/xiaokang_lncRNA/Cleandata/BN2-1060/BN2-1060_R2.fq.gz .
ln -s /data1/linhua/QIANLAB/PROJECT/Long-Noncoding-RNA-project/RAW-data-2016-12-07/xiaokang_lncRNA/Cleandata/BN2-1060/BN2-1060_R1.fq.gz .
ln -s /data1/linhua/0803-LncRNA-WXK-2-JC/Cleandata/BN2-638/BN2-638_R1.fq.gz .
ln -s /data1/linhua/0803-LncRNA-WXK-2-JC/Cleandata/BN2-638/BN2-638_R2.fq.gz .
ln -s /data1/linhua/0803-LncRNA-WXK-2-JC/Cleandata/BN2-637/BN2-637_R1.fq.gz .
ln -s /data1/linhua/0803-LncRNA-WXK-2-JC/Cleandata/BN2-637/BN2-637_R2.fq.gz .

# trim

#!/bin/bash
cutadapt -u 10 -o Trimmed_BN2-1057_R1.fq.gz  BN2-1057_R1.fq.gz &
cutadapt -u 10 -o Trimmed_BN2-1057_R2.fq.gz  BN2-1057_R2.fq.gz &
cutadapt -u 10 -o Trimmed_BN2-1058_R1.fq.gz  BN2-1058_R1.fq.gz &
cutadapt -u 10 -o Trimmed_BN2-1058_R2.fq.gz  BN2-1058_R2.fq.gz &
cutadapt -u 10 -o Trimmed_BN2-1059_R1.fq.gz  BN2-1059_R1.fq.gz &
cutadapt -u 10 -o Trimmed_BN2-1059_R2.fq.gz  BN2-1059_R2.fq.gz &
cutadapt -u 10 -o Trimmed_BN2-1060_R1.fq.gz  BN2-1060_R1.fq.gz &
cutadapt -u 10 -o Trimmed_BN2-1060_R2.fq.gz  BN2-1060_R2.fq.gz &
cutadapt -u 10 -o Trimmed_BN2-637_R1.fq.gz  BN2-637_R1.fq.gz &
cutadapt -u 10 -o Trimmed_BN2-637_R2.fq.gz  BN2-637_R2.fq.gz &
cutadapt -u 10 -o Trimmed_BN2-638_R1.fq.gz  BN2-638_R1.fq.gz &
cutadapt -u 10 -o Trimmed_BN2-638_R2.fq.gz  BN2-638_R2.fq.gz &
cutadapt -u 10 -o Trimmed_BN2-971_R1.fq.gz  BN2-971_R1.fq.gz &
cutadapt -u 10 -o Trimmed_BN2-971_R2.fq.gz  BN2-971_R2.fq.gz &
cutadapt -u 10 -o Trimmed_BN2-972_R1.fq.gz  BN2-972_R1.fq.gz &
cutadapt -u 10 -o Trimmed_BN2-972_R2.fq.gz  BN2-972_R2.fq.gz &
cutadapt -u 10 -o Trimmed_BN2-973_R1.fq.gz  BN2-973_R1.fq.gz &
cutadapt -u 10 -o Trimmed_BN2-973_R2.fq.gz  BN2-973_R2.fq.gz &

## QC

### 
annotation

## 1. Tophat2 is relied on the bowtie2. So we need to index the fasta files.

bowtie2-build –f Homo_sapiens.GRCh37.74.dna.toplevel.fa GRCh37.74


bowtie2-build –f TAIR10.fa TAIR10


## 2. tophat build index

tophat2 -G Araport11_GFF3_genes_transposons.201606.gtf --transcriptome-index=TAIR10.tr TAIR10





## align
## --------

#!/bin/bash

THREADS=10

GFF="/data1/linhua/QIANLAB/Araport_TOPHAT/TAIR10.tr/Araport11_GFF3_genes_transposons.201606"
REF="/data1/linhua/QIANLAB/Araport_TOPHAT/TAIR10"

ID=$(basename ${1} _R1.fq.gz)

tophat2 \
--min-intron-length 8 \
--max-intron-length 5000 \
-p ${THREADS} \
-r -25 \
--library-type fr-firststrand \
-o ${ID}_tophat_align \
--transcriptome-index ${GFF} \
${REF} \
${ID}_R1.fq.gz ${ID}_R2.fq.gz




$$ batch align

#!/bin/bash
## batch align reads to the reference genome of TAIR10

sh align_tophat2_araport.sh Trimmed_BN2-1057_R1.fq.gz
sh align_tophat2_araport.sh Trimmed_BN2-1058_R1.fq.gz
sh align_tophat2_araport.sh Trimmed_BN2-1059_R1.fq.gz
sh align_tophat2_araport.sh Trimmed_BN2-1060_R1.fq.gz
sh align_tophat2_araport.sh Trimmed_BN2-637_R1.fq.gz
sh align_tophat2_araport.sh Trimmed_BN2-638_R1.fq.gz
sh align_tophat2_araport.sh Trimmed_BN2-971_R1.fq.gz
sh align_tophat2_araport.sh Trimmed_BN2-972_R1.fq.gz
sh align_tophat2_araport.sh Trimmed_BN2-973_R1.fq.gz



## subsample the bam files

#!/bin/bash
sambamba view -h -t 10 -s 0.1 -f bam --subsampling-seed=12345 epi_merge.bam -o 10_epi_merge.bam
sambamba view -h -t 10 -s 0.2 -f bam --subsampling-seed=12345 epi_merge.bam -o 20_epi_merge.bam
sambamba view -h -t 10 -s 0.3 -f bam --subsampling-seed=12345 epi_merge.bam -o 30_epi_merge.bam
sambamba view -h -t 10 -s 0.4 -f bam --subsampling-seed=12345 epi_merge.bam -o 40_epi_merge.bam
sambamba view -h -t 10 -s 0.5 -f bam --subsampling-seed=12345 epi_merge.bam -o 50_epi_merge.bam
sambamba view -h -t 10 -s 0.6 -f bam --subsampling-seed=12345 epi_merge.bam -o 60_epi_merge.bam
sambamba view -h -t 10 -s 0.7 -f bam --subsampling-seed=12345 epi_merge.bam -o 70_epi_merge.bam
sambamba view -h -t 10 -s 0.8 -f bam --subsampling-seed=12345 epi_merge.bam -o 80_epi_merge.bam
sambamba view -h -t 10 -s 0.9 -f bam --subsampling-seed=12345 epi_merge.bam -o 90_epi_merge.bam


## 

#!/bin/bash

sh string_tie_test.sh 10_epi_merge.bam 
sh string_tie_test.sh 20_epi_merge.bam 
sh string_tie_test.sh 30_epi_merge.bam 
sh string_tie_test.sh 40_epi_merge.bam 
sh string_tie_test.sh 50_epi_merge.bam 
sh string_tie_test.sh 60_epi_merge.bam 
sh string_tie_test.sh 70_epi_merge.bam 
sh string_tie_test.sh 80_epi_merge.bam 
sh string_tie_test.sh 90_epi_merge.bam 


## link to a new position
ln -s /data1/linhua/QIANLAB/PROJECT/Long-Noncoding-RNA-project/RE_ANALYSIS_LNC_RNA_DATA_SET/Remapping_LncRNA-Seq/Trimmed_BN2-972_tophat_align/accepted_hits.bam  0972_tophat2.bam
ln -s /data1/linhua/QIANLAB/PROJECT/Long-Noncoding-RNA-project/RE_ANALYSIS_LNC_RNA_DATA_SET/Remapping_LncRNA-Seq/Trimmed_BN2-973_tophat_align/accepted_hits.bam  0973_tophat2.bam
ln -s /data1/linhua/QIANLAB/PROJECT/Long-Noncoding-RNA-project/RE_ANALYSIS_LNC_RNA_DATA_SET/Remapping_LncRNA-Seq/Trimmed_BN2-637_tophat_align/accepted_hits.bam  0637_tophat2.bam
ln -s /data1/linhua/QIANLAB/PROJECT/Long-Noncoding-RNA-project/RE_ANALYSIS_LNC_RNA_DATA_SET/Remapping_LncRNA-Seq/Trimmed_BN2-971_tophat_align/accepted_hits.bam  0971_tophat2.bam
ln -s /data1/linhua/QIANLAB/PROJECT/Long-Noncoding-RNA-project/RE_ANALYSIS_LNC_RNA_DATA_SET/Remapping_LncRNA-Seq/Trimmed_BN2-1059_tophat_align/accepted_hits.bam 1059_tophat2.bam
ln -s /data1/linhua/QIANLAB/PROJECT/Long-Noncoding-RNA-project/RE_ANALYSIS_LNC_RNA_DATA_SET/Remapping_LncRNA-Seq/Trimmed_BN2-1058_tophat_align/accepted_hits.bam 1058_tophat2.bam
ln -s /data1/linhua/QIANLAB/PROJECT/Long-Noncoding-RNA-project/RE_ANALYSIS_LNC_RNA_DATA_SET/Remapping_LncRNA-Seq/Trimmed_BN2-1057_tophat_align/accepted_hits.bam 1057_tophat2.bam
ln -s /data1/linhua/QIANLAB/PROJECT/Long-Noncoding-RNA-project/RE_ANALYSIS_LNC_RNA_DATA_SET/Remapping_LncRNA-Seq/Trimmed_BN2-638_tophat_align/accepted_hits.bam  0638_tophat2.bam
ln -s /data1/linhua/QIANLAB/PROJECT/Long-Noncoding-RNA-project/RE_ANALYSIS_LNC_RNA_DATA_SET/Remapping_LncRNA-Seq/Trimmed_BN2-1060_tophat_align/accepted_hits.bam 1060_tophat2.bam

#!/bin/bash

## remove the multiple mapping reads and cp and mt reads

echo "index the tophat2 mapped bam files"

sambamba index -t 10 0637_tophat2.bam
sambamba index -t 10 0638_tophat2.bam
sambamba index -t 10 0971_tophat2.bam
sambamba index -t 10 0972_tophat2.bam
sambamba index -t 10 0973_tophat2.bam
sambamba index -t 10 1057_tophat2.bam
sambamba index -t 10 1058_tophat2.bam
sambamba index -t 10 1059_tophat2.bam
sambamba index -t 10 1060_tophat2.bam

echo "Start filter unneccssary reads"

samtools view -@ 2 -h 0637_tophat2.bam Chr1 Chr2 Chr3 Chr4 Chr5|grep -E '^@|NH:i:1' |samtools view -@ 2 -Sb - > uniqmap_0637.bam &
samtools view -@ 2 -h 0638_tophat2.bam Chr1 Chr2 Chr3 Chr4 Chr5|grep -E '^@|NH:i:1' |samtools view -@ 2 -Sb - > uniqmap_0638.bam &
samtools view -@ 2 -h 0971_tophat2.bam Chr1 Chr2 Chr3 Chr4 Chr5|grep -E '^@|NH:i:1' |samtools view -@ 2 -Sb - > uniqmap_0971.bam &
samtools view -@ 2 -h 0972_tophat2.bam Chr1 Chr2 Chr3 Chr4 Chr5|grep -E '^@|NH:i:1' |samtools view -@ 2 -Sb - > uniqmap_0972.bam &
samtools view -@ 2 -h 0973_tophat2.bam Chr1 Chr2 Chr3 Chr4 Chr5|grep -E '^@|NH:i:1' |samtools view -@ 2 -Sb - > uniqmap_0973.bam &
samtools view -@ 2 -h 1057_tophat2.bam Chr1 Chr2 Chr3 Chr4 Chr5|grep -E '^@|NH:i:1' |samtools view -@ 2 -Sb - > uniqmap_1057.bam &
samtools view -@ 2 -h 1058_tophat2.bam Chr1 Chr2 Chr3 Chr4 Chr5|grep -E '^@|NH:i:1' |samtools view -@ 2 -Sb - > uniqmap_1058.bam &
samtools view -@ 2 -h 1059_tophat2.bam Chr1 Chr2 Chr3 Chr4 Chr5|grep -E '^@|NH:i:1' |samtools view -@ 2 -Sb - > uniqmap_1059.bam &
samtools view -@ 2 -h 1060_tophat2.bam Chr1 Chr2 Chr3 Chr4 Chr5|grep -E '^@|NH:i:1' |samtools view -@ 2 -Sb - > uniqmap_1060.bam &

sambamba index -t 10 uniqmap_0637.bam
sambamba index -t 10 uniqmap_0638.bam
sambamba index -t 10 uniqmap_0971.bam
sambamba index -t 10 uniqmap_0972.bam
sambamba index -t 10 uniqmap_0973.bam
sambamba index -t 10 uniqmap_1057.bam
sambamba index -t 10 uniqmap_1058.bam
sambamba index -t 10 uniqmap_1059.bam
sambamba index -t 10 uniqmap_1060.bam

#!/bin/bash
## reads sort by names

samtools sort -n -@ 3 --output-fmt BAM -o names_uniqmap_0637.bam uniqmap_0637.bam &
samtools sort -n -@ 3 --output-fmt BAM -o names_uniqmap_0638.bam uniqmap_0638.bam &
samtools sort -n -@ 3 --output-fmt BAM -o names_uniqmap_0971.bam uniqmap_0971.bam &
samtools sort -n -@ 3 --output-fmt BAM -o names_uniqmap_0972.bam uniqmap_0972.bam &
samtools sort -n -@ 3 --output-fmt BAM -o names_uniqmap_0973.bam uniqmap_0973.bam &
samtools sort -n -@ 3 --output-fmt BAM -o names_uniqmap_1057.bam uniqmap_1057.bam &
samtools sort -n -@ 3 --output-fmt BAM -o names_uniqmap_1058.bam uniqmap_1058.bam &
samtools sort -n -@ 3 --output-fmt BAM -o names_uniqmap_1059.bam uniqmap_1059.bam &
samtools sort -n -@ 3 --output-fmt BAM -o names_uniqmap_1060.bam uniqmap_1060.bam &

## -------

#!/bin/bash

ID=$(basename ${1} .bam)

sambamba index -t 10 ${1}

samtools view -@ 10 -h ${1} Chr1 Chr2 Chr3 Chr4 Chr5|grep -E '^@|NH:i:1' |samtools view -@ 10 -Sb - > uniqmap_${ID}.bam

sambamba index -t 10 uniqmap_${ID}.bam

samtools sort -n -@ 10 --output-fmt BAM -o names_uniqmap_${ID}.bam uniqmap_${ID}.bam

## -------


##

bam2wig.py -s ${CHROM} -i  rdd.bam  -o rdd_wig -u --strand="1+-,1-+,2++,2--"


####

#!/bin/bash
bam2wig.py -s ${CHROM} -i uniqmap_0637.bam -o uniqmap_nCol-0  -u --strand="1+-,1-+,2++,2--"  &
bam2wig.py -s ${CHROM} -i uniqmap_0638.bam -o uniqmap_ddm1  -u --strand="1+-,1-+,2++,2--"  &
bam2wig.py -s ${CHROM} -i uniqmap_0971.bam -o uniqmap_nrpd1   -u --strand="1+-,1-+,2++,2--"  &
bam2wig.py -s ${CHROM} -i uniqmap_0972.bam -o uniqmap_nrpe1  -u --strand="1+-,1-+,2++,2--"  &
bam2wig.py -s ${CHROM} -i uniqmap_0973.bam -o uniqmap_morc6  -u --strand="1+-,1-+,2++,2--"  &
bam2wig.py -s ${CHROM} -i uniqmap_1057.bam -o uniqmap_rdd  -u --strand="1+-,1-+,2++,2--"  &
bam2wig.py -s ${CHROM} -i uniqmap_1058.bam -o uniqmap_met1  -u --strand="1+-,1-+,2++,2--"  &
bam2wig.py -s ${CHROM} -i uniqmap_1059.bam -o uniqmap_mom1  -u --strand="1+-,1-+,2++,2--"  &
bam2wig.py -s ${CHROM} -i uniqmap_1060.bam -o uniqmap_cCol-0  -u --strand="1+-,1-+,2++,2--"  &



## reannotation 

#!/bin/bash
sh string_tie_test.sh uniqmap_0637.bam &
sh string_tie_test.sh uniqmap_0638.bam &
sh string_tie_test.sh uniqmap_0971.bam &
sh string_tie_test.sh uniqmap_0972.bam &
sh string_tie_test.sh uniqmap_0973.bam &
sh string_tie_test.sh uniqmap_1057.bam &
sh string_tie_test.sh uniqmap_1058.bam &
sh string_tie_test.sh uniqmap_1059.bam &
sh string_tie_test.sh uniqmap_1060.bam &



##
## merge multiple bam files
## except the 1060 (cytoplasm)

nohup sambamba merge --nthreads=20 epi_merge.bam uniqmap_0637.bam \
uniqmap_0638.bam \
uniqmap_0971.bam \
uniqmap_0972.bam \
uniqmap_0973.bam \
uniqmap_1057.bam \
uniqmap_1058.bam \
uniqmap_1059.bam &






####################################################################################################

#》》》》 creat a scripts : check_qc_multi_gtfs.sh

#!/bin/bash

GFF="/data1/linhua/QIANLAB/Araport_related_data/Araport11_GFF3_genes_transposons.201606.gff"
ID=$(basename ${1} .gtf)

gffcompare -R -r ${GFF} -o gtf_quality_${ID} ${1}

#!/bin/bash

## eval_gtf.sh is in (/data1/linhua/software/BIN/eval_gtf.sh)

eval_gtf.sh  f_0.gtf &
eval_gtf.sh  f_10.gtf &
eval_gtf.sh  f_20.gtf &
eval_gtf.sh  f_30.gtf &
eval_gtf.sh  f_40.gtf &
eval_gtf.sh  f_50.gtf &
eval_gtf.sh  f_60.gtf &
eval_gtf.sh  f_70.gtf &
eval_gtf.sh  f_80.gtf &
eval_gtf.sh  f_90.gtf &
eval_gtf.sh  f_95.gtf 

grep "Base level" 			*stats	|sed "s/://g;s/Base level//g"     		|tr "|" " "    > Base_level.summary
grep "Exon level" 			*stats		 |sed "s/://g;s/Exon level//g"     		|tr "|" " "    > Exon_level.summary
grep "Intron level" 		*stats	|sed "s/://g;s/Intron level//g"     	|tr "|" " "    > Intron_level.summary
grep "Intron chain level"   *stats  |sed "s/://g;s/Intron chain level//g"  |tr "|" " "    > Intron_chain_level.summary
grep "Transcript level"     *stats    |sed "s/://g;s/Transcript level//g"    |tr "|" " "    > Transcript_level.summary
grep "Locus level"          *stats   		 |sed "s/://g;s/Locus level//g"     	|tr "|" " "    > Locus_level.summary

## /data1/linhua/software/BIN/plot_GTF_QUAL.R

plot_GTF_QUAL.R

####################################################################################################

#!/bin/bash


lOC="/data1/linhua/QIANLAB/PROJECT/Long-Noncoding-RNA-project/LNC-2016-12-08/DIFF/reassembley_by_stringtie/CHOOSE_TWO/Test_on_seq_depth"
cd ${LOC}/Sim_30_epi_merge_stringtie_out_2016-12-26 && check_qc_multi_gtfs.sh
cd ${LOC}/Sim_40_epi_merge_stringtie_out_2016-12-26 && check_qc_multi_gtfs.sh
cd ${LOC}/Sim_50_epi_merge_stringtie_out_2016-12-26 && check_qc_multi_gtfs.sh
cd ${LOC}/Sim_60_epi_merge_stringtie_out_2016-12-27 && check_qc_multi_gtfs.sh
cd ${LOC}/Sim_70_epi_merge_stringtie_out_2016-12-27 && check_qc_multi_gtfs.sh
cd ${LOC}/Sim_80_epi_merge_stringtie_out_2016-12-27 && check_qc_multi_gtfs.sh
cd ${LOC}/Sim_90_epi_merge_stringtie_out_2016-12-27 && check_qc_multi_gtfs.sh
cd ${LOC}/Sim_10_epi_merge_stringtie_out_2016-12-26 && check_qc_multi_gtfs.sh
cd ${LOC}/Sim_20_epi_merge_stringtie_out_2016-12-26 && check_qc_multi_gtfs.sh



#!/bin/bash

## changed_tx_qual.R:/data1/linhua/software/BIN/changed_tx_qual.R

changed_tx_qual.R 







cut -f2 f_0.gtf_compare_ara_stats.tsv >   f_0_2.txxt
cut -f2 f_10.gtf_compare_ara_stats.tsv > f_10_2.txxt
cut -f2 f_20.gtf_compare_ara_stats.tsv > f_20_2.txxt
cut -f2 f_30.gtf_compare_ara_stats.tsv > f_30_2.txxt
cut -f2 f_40.gtf_compare_ara_stats.tsv > f_40_2.txxt
cut -f2 f_50.gtf_compare_ara_stats.tsv > f_50_2.txxt
cut -f2 f_60.gtf_compare_ara_stats.tsv > f_60_2.txxt
cut -f2 f_70.gtf_compare_ara_stats.tsv > f_70_2.txxt
cut -f2 f_80.gtf_compare_ara_stats.tsv > f_80_2.txxt
cut -f2 f_90.gtf_compare_ara_stats.tsv > f_90_2.txxt
cut -f2 f_95.gtf_compare_ara_stats.tsv > f_95_2.txxt

paste f_0_2.txxt \
f_10_2.txxt \
f_20_2.txxt \
f_30_2.txxt \
f_40_2.txxt \
f_50_2.txxt \
f_60_2.txxt \
f_70_2.txxt \
f_80_2.txxt \
f_90_2.txxt \
f_95_2.txxt > f_summary_stats.tsv

